proposals/0000-even-more-improved-integers.md

```diff
@@ -0,0 +1,1107 @@
+  /// For example:
+  ///
+  ///     var x = 15
+  ///     y *= 7
+  ///     // y == 105
+  static func *=(_ lhs: inout Self, rhs: Self)
+}
+
+extension Number {
+  public static prefix func + (x: Self) -> Self {
+    return x
+  }
+}
+```
+
+#### `SignedNumber`
+
+The `SignedNumber` protocol is for numbers that can be negated.
+
+```Swift
+public protocol SignedNumber : Number {
+  /// Returns the additive inverse of this value.
+  ///
+  ///     let x = 21
+  ///     let y = -x
+  ///     // y == -21
+  ///
+  /// - Returns: The additive inverse of this value.
+  ///
+  /// - SeeAlso: `negate()`
+  static prefix func - (_ operand: Self) -> Self
+
+  /// Replaces this value with its additive inverse.
+  ///
+  /// The following example uses the `negate()` method to negate the value of
+  /// an integer `x`:
+  ///
+  ///     var x = 21
+  ///     x.negate()
+  ///     // x == -21
+  ///
+  /// - SeeAlso: The unary minus operator (`-`).
+  mutating func negate()
+}
+
+extension SignedNumber {
+  public static prefix func - (_ operand: Self) -> Self {
+    var result = operand
+    result.negate()
+    return result
+  }
+
+  public mutating func negate() {
+    self = 0 - self
+  }
+}
+```
+
+#### `BinaryInteger`
+
+The `BinaryInteger` protocol is the basis for all the integer types provided by
+the standard library.
+
+This protocol adds a few new initializers. Two of them allow to create integers
+from floating point numbers, others support construction from instances of any
+type conforming to `BinaryInteger`, using different strategies:
+
+  - Initialize `Self` with the value, provided that the value is representable.
+    The precondition should be satisfied by the caller.
+
+  - Extend or truncate the value to fit into `Self`
+
+  - Clamp the value to the representable range of `Self`
+
+`BinaryInteger` also declares bitwise and shift operators.
+
+```Swift
+public protocol BinaryInteger :
+  Comparable, Hashable, Number, CustomStringConvertible, Strideable {
+
+  associatedtype Words : Collection // where Iterator.Element == UInt
+
+  /// A Boolean value indicating whether this type is a signed integer type.
+  ///
+  /// *Signed* integer types can represent both positive and negative values.
+  /// *Unsigned* integer types can represent only nonnegative values.
+  static var isSigned: Bool { get }
+
+  /// Creates an integer from the given floating-point value, if it can be
+  /// represented exactly.
+  ///
+  /// If the value passed as `source` is not representable exactly, the result
+  /// is `nil`. In the following example, the constant `x` is successfully
+  /// created from a value of `21.0`, while the attempt to initialize the
+  /// constant `y` from `21.5` fails:
+  ///
+  ///     let x = Int(exactly: 21.0)
+  ///     // x == Optional(21)
+  ///     let y = Int(exactly: 21.5)
+  ///     // y == nil
+  ///
+  /// - Parameter source: A floating-point value to convert to an integer.
+  init?<T : FloatingPoint>(exactly source: T)
```

Comment:
Can this initializer be moved to the `Number` protocol? Or will it be duplicated in the `BinaryFloatingPoint` or `FloatingPoint` protocols?